mesonpy/__init__.py

```diff
@@ -704,6 +706,15 @@ def _validate_metadata(self) -> None:
                     f'expected `{self._metadata.requires_python}`'
                 )
 
+    def _check_for_unknown_config_keys(self) -> None:
```

Comment:
Design thing: this kind of code tends to be easier to understand if the dict/keys to validate are passed in explicitly, rather than magically fetched. 

I was originally making this comment for `self._meson_args`, however `(args,)` two lines below is worse - that's neither passed in nor created in this function, which is a worse problem (you need to go look for where it came from, plus you cannot safely refactor this code).

Reply:
I did some refactoring, making this

```python
self._check_for_unknown_config_keys({
    'args': typing_get_args(MesonArgsKeys),
 })
```

Where `typing_get_args(MesonArgsKeys)` are the valid `args` keys -- they are extracted from the `MesonArgsKeys` type, so that we have a single source of truth for them.

Let me know what you think.